infra/cmake/modules/ExternalSourceTools.cmake

```diff
@@ -103,7 +103,13 @@ function(ExternalSource_Download PREFIX)
 
     message(STATUS "Extract ${PREFIX}")
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz "${DOWNLOAD_PATH}"
-                    WORKING_DIRECTORY "${TMP_DIR}")
+                    WORKING_DIRECTORY "${TMP_DIR}"
+                    ERROR_VARIABLE EXTRACTION_ERROR)
+
+    if(EXTRACTION_ERROR)
+      message(FATAL_ERROR "Extract ${PREFIX} - failed")
```

Comment:
Can you please provide an sample output you got?

Reply:
From https://github.com/Samsung/ONE/pull/7786
> As a regression(actually more like early-fail) for https://github.com/Samsung/ONE/issues/7783
> 
> I think it should stop on configure stage not build stage
> 
> Currently,
> on configure, just skip
> ```
> -- Download EIGEN from https://gitlab.com/libeigen/eigen/-/archive/386d809bde475c65b7940f290efe80e6a05878c4/eigen-386d809bde475c65b7940f290efe80e6a05878c4.tar.gz
> -- (Trial Count : 0)
> -- Download EIGEN from https://gitlab.com/libeigen/eigen/-/archive/386d809bde475c65b7940f290efe80e6a05878c4/eigen-386d809bde475c65b7940f290efe80e6a05878c4.tar.gz - done
> -- Extract EIGEN
> CMake Error: Problem with archive_read_open_file(): Unrecognized archive format
> CMake Error: Problem extracting tar: /home/circleci/project/deps/NNAS/externals/TENSORFLOW-2.3.0-EIGEN-eigen-386d809bde475c65b7940f290efe80e6a05878c4.tar.gz
> -- Extract EIGEN - done
> ```
> 
> on build, it failed to find source code
> ```
>   #include "Eigen/Core"
>           ^~~~~~~~~~~~
> compilation terminated. 
> ```

Reply:
(I didn't do negative test)

Reply:
I tried `RESULT_VARIABLE` in #7786, not `ERROR_VARIABLE`. I guess both works as expected..? :smile:

Reply:
> Can you please provide an sample output you got? You can paste in in a comment of this PR.

Done! Corresponding comment added to PR comment =)